json-reporter.js

```diff
@@ -0,0 +1,30 @@
+class MyCustomReporter {
+  constructor(globalConfig, options) {
+    this._globalConfig = globalConfig;
+    this._options = options;
+  }
+
+  onRunComplete(contexts, results) {
+    results.testResults[0].testResults.forEach((item) => {
+      if (item.status !== "failed") {
+        return;
+      }
+      const newLine = "%0A";
+      const name = results.testResults[0].testFilePath;
+
+      if (!item.location) {
+        throw new Error(
+          "You need to invoked jest with argument --testLocationInResults"
+        );
+      }
+
+      const line = item.location.line + 1;
```

Comment:
`+ 1`? `testLocation` is for the `it`, not the `expect`. You're probably better of looking at the stack trace in the error

Reply:
I misread the documentation 

> Note that column is 0-indexed while line is not.
https://jestjs.io/docs/en/cli.html#--testlocationinresults

Anyway, thanks for the hint about the `it` this wasn't clear from the documentation. Will parse the error message as you suggested.

Reply:
I updated the implementation according to your comment. Should I try to get a Jest PR up? If you could point me to a location / package where I should make changes, that would be amazing. From a quick look it feels like it should be part of https://github.com/facebook/jest/tree/master/packages/jest-reporters/

Reply:
Part of reporters, yeah.

Auto-adding it would go somewhere here: https://github.com/facebook/jest/blob/faa52673f4d02ee1d273103ada9382a0805b54a4/packages/jest-core/src/TestScheduler.ts#L267-L298

We should probably wait a bit with that until we're happy, though

Reply:
I'm already super happy with the result 😉 

I want to get this feature in so I can deprecate a [Danger.js](https://github.com/danger/danger-js) plugin I wrote to present failing tests on a Pull Request. Also I think the Jest community will love this integration which works out of the box with every GitHub Action setup.

![image](https://user-images.githubusercontent.com/1393946/80655043-030e4280-8a7e-11ea-956f-eac980dae08b.png)